adapters/alkimi/alkimi.go

```diff
@@ -0,0 +1,194 @@
+package alkimi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/floors"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+const price_macro = "${AUCTION_PRICE}"
+
+type adapter struct {
+	endpoint string
+}
+
+type extObj struct {
+	AlkimiBidderExt openrtb_ext.ExtImpAlkimi `json:"bidder"`
+}
+
+// Builder builds a new instance of the Alkimi adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	endpointURL, err := url.Parse(config.Endpoint)
+	if err != nil || len(endpointURL.String()) == 0 {
+		return nil, fmt.Errorf("invalid endpoint: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: endpointURL.String(),
+	}
+	return bidder, nil
+}
+
+// MakeRequests creates Alkimi adapter requests
+func (adapter *adapter) MakeRequests(request *openrtb2.BidRequest, req *adapters.ExtraRequestInfo) (reqsBidder []*adapters.RequestData, errs []error) {
+	reqCopy := *request
+
+	updated, errs := updateImps(reqCopy)
+	if len(errs) > 0 || len(reqCopy.Imp) != len(updated) {
+		return nil, errs
+	}
+
+	reqCopy.Imp = updated
+	encoded, err := json.Marshal(reqCopy)
+	if err != nil {
+		errs = append(errs, err)
+	} else {
+		reqBidder := buildBidderRequest(adapter, encoded)
+		reqsBidder = append(reqsBidder, reqBidder)
+	}
+	return
+}
+
+func updateImps(bidRequest openrtb2.BidRequest) ([]openrtb2.Imp, []error) {
+	var errs []error
+
+	updatedImps := make([]openrtb2.Imp, 0, len(bidRequest.Imp))
+	for _, imp := range bidRequest.Imp {
+
+		var bidderExt adapters.ExtImpBidder
+		var extImpAlkimi openrtb_ext.ExtImpAlkimi
+
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := json.Unmarshal(bidderExt.Bidder, &extImpAlkimi); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		var bidFloorPrice floors.Price
+		bidFloorPrice.FloorMinCur = imp.BidFloorCur
+		bidFloorPrice.FloorMin = imp.BidFloor
+
+		if len(bidFloorPrice.FloorMinCur) > 0 && bidFloorPrice.FloorMin > 0 {
+			imp.BidFloor = bidFloorPrice.FloorMin
+		} else {
+			imp.BidFloor = extImpAlkimi.BidFloor
+		}
+		imp.Instl = extImpAlkimi.Instl
```

Comment:
Note that. As per static/bidder-params/alkimi.json, `extImpAlkimi.Instl` is not required param so can be empty string and imp.Instl will have in empty string value